aws/resource_aws_elasticsearch_domain.go

```diff
@@ -431,6 +504,17 @@ func resourceAwsElasticSearchDomainUpdate(d *schema.ResourceData, meta interface
 		}
 	}
 
+	if d.HasChange("vpc_options") {
+		options := d.Get("vpc_options").([]interface{})
+
+		if len(options) > 1 {
+			return fmt.Errorf("Only a single vpc_options block is expected")
```

Comment:
We can do the same validation just by adding `MaxItems: 1` to the field in the schema which is a bit more idiomatic and more importantly the user will get the validation error during plan, rather than apply.